api/src/main/java/org/pmiops/workbench/interceptors/AuthInterceptor.java

```diff
@@ -118,8 +120,22 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
     // TODO: check Google group membership to ensure user is in registered user group
 
     String userEmail = userInfo.getEmail();
+    WorkbenchConfig workbenchConfig = workbenchConfigProvider.get();
+    if (workbenchConfig.auth.serviceAccountApiUsers.contains(userEmail)) {
+      // Whitelisted service accounts are able to make API calls, too.
+      // TODO: stop treating service accounts as normal users, have a separate table for them,
+      // administrators.
+      User user = userDao.findUserByEmail(userEmail);
+      if (user == null) {
+        user = userService.createUser(null, null, userEmail, null);
```

Comment:
Is there anything preventing us from attempting to register this service account as a Firecloud user? Would that work (and if not, what are the implications of having a workbench user that isn't an FC user)?

Reply:
You can register service accounts as Firecloud users. We already do this for our AppEngine service accounts as a part of environment setup, so there's no need to worry about it here... in general I'd say that any service accounts that are making FC API calls should have been registered with FC independently from our service.